src/components/SearchAndFilter/SearchAndFilter.js

```diff
@@ -13,14 +18,56 @@ const SearchAndFilter = ({ externallyControlled = false, onChange }) => {
     }
   };
 
+  // This useEffect sets up listeners so the panel will close if user clicks
+  // anywhere else on the page or hits the escape key
+  useEffect(() => {
+    const closePanel = () => {
+      setFilterPanelVisible(false);
+    };
+
+    const mouseDown = (e) => {
+      // Check if click is outside of filter panel
+      if (!filterPanelRef.current.contains(e.target)) {
+        // If so, close the panel
+        closePanel();
+      }
+    };
+
+    const keyDown = (e) => {
+      if (e.code === "Escape") {
+        // Close panel if Esc keydown detected
+        closePanel();
+      }
+    };
+
+    // Add listener on document to capture click events
+    document.addEventListener("mousedown", mouseDown);
+    // Add listener on document to capture keydown events
+    document.addEventListener("keydown", keyDown);
+    // return function to be called when unmounted
+    return () => {
+      document.removeEventListener("mousedown", mouseDown);
+      document.removeEventListener("keydown", keyDown);
+    };
+  }, []);
+
   return (
     <div className="search-and-filter">
       <SearchBox
+        autocomplete="off"
         externallyControlled={externallyControlled}
         placeholder="Search and filter"
         onChange={(searchTerm) => searchOnChange(searchTerm)}
+        onFocus={() => setFilterPanelVisible(true)}
         value={searchTerm}
       />
+      <div
+        className="search-and-filter__panel"
+        ref={filterPanelRef}
+        data-visible={filterPanelVisible}
```

Comment:
Why not use `aria-hidden` and make it accessible from the get go?

Reply:
Good call 🙌🏻